packages/editor/src/components/block-list/block-draggable.js

```diff
@@ -2,30 +2,57 @@
  * External dependencies
  */
 import classnames from 'classnames';
+import { noop } from 'lodash';
 
 /**
  * WordPress dependencies
  */
-import { Draggable } from '@wordpress/components';
+import { Component } from '@wordpress/element';
+import { withDraggable } from '@wordpress/components';
+import { withSafeTimeout } from '@wordpress/compose';
 
-function BlockDraggable( { rootClientId, index, clientId, layout, isDragging, ...props } ) {
-	const className = classnames( 'editor-block-list__block-draggable', {
-		'is-visible': isDragging,
-	} );
+class BlockDraggable extends Component {
+	constructor( ) {
+		super( ...arguments );
+		this.onDragStart = this.onDragStart.bind( this );
+		this.onDragEnd = this.onDragEnd.bind( this );
+	}
 
-	const transferData = {
-		type: 'block',
-		fromIndex: index,
-		rootClientId,
-		clientId,
-		layout,
-	};
+	onDragStart( event ) {
+		const { elementId, rootClientId, clientId, layout, index, setTimeout, initDragging, onDragStart = noop } = this.props;
+		const transferData = {
+			type: 'block',
+			fromIndex: index,
+			rootClientId,
+			clientId,
+			layout,
+		};
+		initDragging( elementId, transferData )( event );
+		setTimeout( () => onDragStart( event ) );
```

Comment:
Note that we need to make sure `onDragStart` is executed after `initDragging`.

This is due to how `Draggable` and Drag and Drop behavior was originally [set up](https://github.com/WordPress/gutenberg/blob/5b81cbc66c5d52f9603f77a60ec7864150173dd0/packages/components/src/draggable/index.js#L130). The `isDragging` prop takes its value from `state.dragging` which is [changed in the `onDragStart` and `onDragEnd` methods](https://github.com/WordPress/gutenberg/blob/5b81cbc66c5d52f9603f77a60ec7864150173dd0/packages/editor/src/components/block-list/block.js#L342) of the block. Upon `isDragging` being true, [the visibility of any block's children is changed to `hidden`](https://github.com/WordPress/gutenberg/blob/5b81cbc66c5d52f9603f77a60ec7864150173dd0/packages/editor/src/components/block-list/style.scss#L95), making them un-reactive to events (so the `initDragging` handler ends up not being set up).